File: ret/Task.java

public class Task {

  private String name;
  private String description;
  private Integer percent;
  private Integer level;

  public Task() {
      name = "";
      description = "";
      percent = 0;
      level = 0;

  }

  public String getName() {
      return this.name;
  }

  public void setName(String input) {
      this.name = input;
  }

  public String getDescription() {
      return this.description;
  }

  public void setDescription(String input) {
      this.description = input;
  }

  public Integer getPercent() {
      return this.percent;
  }

  public void setPercent(Integer input) {
      this.percent = input;
  }

  public Integer getLevel() {
      return this.level;
  }

  public void setLevel(Integer input) {
      this.level = input;
  }

}

File: ret/task.py


class task:
    def __init__(self, level=0):
        self.name = ''
        self.description = ''
        self.percent = 0
        self.hours = None
        self.level = level
        self.subtasks = []

    def add_subtask(self, oTask):
        if self.subtasks is None:
            self.subtasks = []
        oTask.level = self.level + 1
        self.subtasks.append(oTask)

    def update_hours(self, lHours):
        self.hours = []
        for iHour in lHours:
            self.hours.append((self.percent/100) * iHour)

    def get_all_tasks(self):
        lReturn = []
        lReturn.append(self)
        for oSubtask in self.subtasks:
            lReturn.extend(oSubtask.get_all_tasks())
        return lReturn
